### PR TITLE
Technology

### LINKED ISSUE #2: Technology class and sub classes
This issue can be closed when ``technology.py`` includes a class called ``Technology`` with the following attributes:

- [ ] region name
- [ ] technology name
- [ ] technology label
- [ ] capacity units
- [ ] input commodity
- [ ] efficiency
- [ ] output commodity
- [ ] capacity to activity
- [ ] capacity factor
- [ ] lifetime
- [ ] lifetimeloan
- [ ] existing capacity

Finally, the class must have corresponding tests.

### PR DESCRIPTION
**Pull request by samgdotson**
This PR adds the following tables to ``db_creator.py``. This PR closes #2. 

* ``create_sectors``
* ``create_technology_labels``
* ``create_technologies``

This PR may appear large, however the important differences are in ``db_creator.py``, ``driver.py``, and ``model_info.py``. 

In ``driver.py`` I added a function ``_collect_technologies`` which gathers all of the initialized technologies in the pygenesys input file. The motivation for this is to avoid having users declare a technology and then asking them to add it to a list in the input file. I.e. the workflow considered previously was:
1. Declare a technology
```py
my_tech = Technology(...)
# or 
from pygenesys.technology.supply import my_tech
```
2. Add that technology to a list or dictionary of all technologies
```py
technologies = [my_tech]
```
3. ``driver.py`` passes ``infile.technologies`` to a ``ModelInfo`` instance.  

The new workflow is
1. Declare the technology as before
2. ``driver.py`` collects the ``Technology`` objects from the input file and passes the list to a ``ModelInfo`` instance.

This is desired because it reduces the number of required steps for users. 

**NOTE** 
In next PR, region dependent parameters and tables will be added. Users will be required to specify the parameters for each region (e.g. efficiency). The method ``add_tech_data`` will be completed next. It was introduced but left incomplete in this PR to keep it small. An associated issue will be created.

### CODE REVIEW

File: examples/example_inputfile.py [hunk 1]
```diff
@@ -53,9 +55,22 @@
                             n_seasons=N_seasons,
                             n_hours=N_hours)
 
-from pygenesys.commodity.resource import electricity, steam
+
+# Add technologies
```
Comment (nsryan2) [inline examples/example_inputfile.py:59]: Isn't this what line 11 does?

**Review by nsryan2 — COMMENTED** [review]
I was able to get everything to work, and the instructions were good except for one:
The execution instructions in example_inputfile.py should be changed to read:
```bash
genesys --infile example_inputfile.py
```
(It wasn't done by this PR, otherwise I'd have made it a suggestion)

File: pygenesys/technology/technology.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        return (f"{self._type}" +
+                f"(\"{self.tech_name}\"," +
+                f"\"{self.tech_label}\"," +
+                f"\"{self.units}\")")
+
+    def _db_entry(self):
+        return (self.tech_name,
+                self.tech_label,
+                self.tech_sector,
+                self.description + ", " + self.units,
```
[the remark this answers is not in the document]
Comment (samgdotson) [reply pygenesys/technology/technology.py:98]: Db_entry writes a tuple that is specific to a data table in Temoa. Your suggestion is functionally different than what I wrote. By doing this you are adding a new entry in the tuple. Description and units are, atm, both strings and together they are a "description" of the technology. They are in fact one item, not two.

**Comment by samgdotson** [comment]
Thanks @nsryan2, I made the change to the input instructions.

**Pull request merged**